core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/MultiCloudWorkspaceService.scala

```diff
@@ -231,14 +230,12 @@ class MultiCloudWorkspaceService(ctx: RawlsRequestContext,
       azureRelayCreateResult <- traceWithParent("createAzureRelayInWSM", parentContext)(_ =>
         Future(workspaceManagerDAO.createAzureRelay(workspaceId, workspaceRequest.region, ctx))
       )
-      // Create storage account before polling on relay because it takes ~45 seconds to create a relay
```

Comment:
If these calls are parallelize-able, you can start them separately by storing the future, instead of the value, and then assigning the value after all the calls are complete. 
Sorry if my words are coming out as borderline gibberish right now. Hopefully a contrived example is clearer:
```scala
for {
... 
  firstCall: Future[ResultOfFirstCall] = call1()  // since you aren't doing anything with the result yet, this won't block
  secondCall: Future[ResultOfSecondCall] = call2() // this can start without waiting on the first call
  firstResult: ResultOfFirstCall  <- firstCall  // now we have to wait for the first call to complete
  secondResult: ResultOfSecondCall <- secondCall
...
```

This is strictly an optimization, though: the current code will work just fine as-is. I just figured it might be useful, given the comments.

Reply:
Thanks for the suggestion! In this case it isn't necessary because I was just delaying the start of polling for the relay completion until I had finished the creation of the storage container because the relay is known to be the long straw (we already kicked off the relay above). And the relay code will be going away shortly… hopefully in the next couple of weeks.

But this is a good thing for me to keep in mind in the future, so thank you for pointing it out!